git-service/src/softserve/handlers.rs

```diff
@@ -1,32 +1,103 @@
-use super::models::{
-    CreateTokenRequest, CreateTokenResponse, CreateUserRequest, CreateUserResponse,
-};
-use super::{create_token, create_user};
-use actix_web::{web, HttpResponse, Responder};
-
-pub async fn handle_create_user(user: web::Json<CreateUserRequest>) -> impl Responder {
-    match create_user(&user.username) {
-        Ok(message) => HttpResponse::Ok().json(CreateUserResponse {
-            username: user.username.clone(),
-            message,
-        }),
-        Err(e) => HttpResponse::InternalServerError().body(format!("Error: {}", e)),
+use super::errors::GitServiceError;
+use super::models::{CreateRepoRequest, CreateTokenRequest, CreateUserRequest};
+use super::ssh::execute_command;
+use super::{create_repo, create_token, create_user};
+use actix_web::{web, HttpResponse, Result};
+use serde_json::json;
+use tracing::{error, info, warn};
+
+pub async fn handle_create_user(
+    user: web::Json<CreateUserRequest>,
+) -> Result<HttpResponse, GitServiceError> {
+    if check_user_exists(&user.username)? {
+        warn!("Attempt to create existing user: {}", user.username);
```

Comment:
This log should be an error and not a warning since it returns an error
```suggestion
        error!("Attempt to create an existing user: {}", user.username);
```